eradiate/scenes/biosphere.py

```diff
@@ -308,110 +642,134 @@ def _create_leaf_cloud(
+            "specifying the locations of the leaf cloud inside the canopy.",
+        type="list(tuple(:class:`.LeafCloud`, list(list())))",
+        default="[]",
     )
 
-    leaf_radius = documented(
-        attrib_quantity(
-            default=ureg.Quantity(0.1, ureg.m),
-            validator=validator_is_positive,
-            units_compatible=cdu.generator("length"),
-        ),
-        doc="Leaf radius.\n"
-            "\n"
-            "Unit-enabled field (default unit: cdu[length]).",
-        type="float",
-        default="0.1 m",
-    )
     # fmt: on
 
-    @leaf_positions.validator
-    @leaf_orientations.validator
-    def _positions_orientations_validator(self, attribute, value):
-        if not len(self.leaf_positions) == len(self.leaf_orientations):
-            raise ValueError(
-                f"While validating {attribute}:"
-                f"leaf_positions and leaf_orientations must have"
-                f"the same length!"
-                f"Got positions: {len(self.leaf_positions)},"
-                f"orientations: {len(self.leaf_orientations)}."
-            )
+    def bsdfs(self):
+        returndict = {}
+        for crown in self.tree_crowns:
+            returndict = {
+                **returndict,
+                **crown.bsdfs()
+            }
+        return returndict
+
+    def shapes(self, ref=False):
+        returndict = {}
+        for crown in self.tree_crowns:
+            returndict = {
+                **returndict,
+                **crown.shapes(ref)
+            }
+        return returndict
+
+    def kernel_dict(self, ref=True):
+        tree_crown_ids = [crown.id for crown in self.tree_crowns]
+        kernel_dict = {}
+
+        for tree_crown_id in tree_crown_ids:
+            if not ref:
+                kernel_dict[tree_crown_id] = self.shapes(ref=False)[
+                    tree_crown_id]
+            else:
+                kernel_dict[f"bsdf_{tree_crown_id}"] = \
+                    self.bsdfs()[f"bsdf_{tree_crown_id}"]
+                kernel_dict[tree_crown_id] = self.shapes(ref=True)[
+                    tree_crown_id]
+
+        return kernel_dict
+
+
+@BiosphereFactory.register("homogeneous_discrete_canopy")
+@parse_docs
+@attr.s
+class HomogeneousDiscreteCanopy(DiscreteCanopy):
+    """A generator for the `homogenous discrete canopy used in the RAMI benchmark <https://rami-benchmark.jrc.ec.europa.eu/_www/phase/phase_exp.php?strTag=level3&strNext=meas&strPhase=RAMI3&strTagValue=HOM_SOL_DIS>`_.
+
+    The leaf cloud in this canopy can be create in two ways. Please refer to
+    :class:`~eradiate.scenes.biosphere.LeafCloud` for the details.
+    """
+
+    id = documented(
+        attr.ib(
+            default="homogeneous_discrete_canopy",
+            validator=attr.validators.optional(attr.validators.instance_of(str)),
+        ),
+        doc=get_doc(Canopy, "id", "doc"),
+        type=get_doc(Canopy, "id", "type"),
+        default="homogeneous_discrete_canopy",
+    )
```

Comment:
Just an opinion: You actually don't need to overload this attribute (the default `canopy` ID should actually be enough to identify this element in the scene).